AdoptAWidget: SnackBar

Thesmader opened the pull request:
## Description

Adds two dartpad samples for SnackBar documentation:
1. Simple SnackBar usage with an action button.
2. Custom SnackBar using behavior, shape, padding and width.

This pull request is:
- [x] A code sample
- [ ] More references
- [ ] More explanation

## Related Issues

Closes #69490 

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

<!-- Links -->
[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

Thesmader committed:
Update SnackBar API Docs with code samples.

Adds two dartpad samples for SnackBar documnetation:
1. Simple SnackBar usage with action button.
2. Custom SnackBar using behavior, shape, padding and width.

Thesmader committed:
Remove trailing spaces.

shihaohong commented on packages/flutter/lib/src/material/snack_bar.dart:
````diff
@@ -158,6 +158,90 @@ class _SnackBarActionState extends State<SnackBarAction> {
+/// [SnackBarAction].
+///
+/// ```dart preamble
+/// class _ShowSnackBarButton extends StatelessWidget {
+///   @override
+///   Widget build(BuildContext context) {
+///     return ElevatedButton(
+///       child: Text("Show Snackbar"),
+///       onPressed: () {
+///         Scaffold.of(context).showSnackBar(
+///           SnackBar(
+///             content: Text("Awesome Snackbar!"),
+///             action: SnackBarAction(
+///               label: "Action",
+///               onPressed: () {
+///                 // Code to execute
+///               }
+///             ),
+///           ),
+///         );
+///       },
+///     );
+///   }
+/// }
````
It seems unnecessary to refactor this into its own stateless widget. Couldn't you just put this content into the `build` function? Here and in the next example.

Thesmader replied:
I'll refactor the ```_ShowSnackBarButton``` into the build function. For the next example do you want me to remove ```_customSnackBar()``` and place it directly inside the ```showSnackBar()``` call inside ```build()```?

shihaohong replied:
Yeah, I think that makes more sense. My rationale is that we usually separate out a new widget class only when it's being reused, and in this case, it isn't since it's the only instance in the example :)

Thesmader replied:
Alright. I'm working on it.

Thesmader replied:
I have made the required changes. Please review.

yash1200 commented on packages/flutter/lib/src/material/snack_bar.dart:
```diff
@@ -158,6 +158,90 @@ class _SnackBarActionState extends State<SnackBarAction> {
+///     return ElevatedButton(
+///       child: Text("Show Snackbar"),
+///       onPressed: () {
+///         Scaffold.of(context).showSnackBar(
```
Use ScaffoldMessenger instead of Scaffold to call showSnackBar function.

shihaohong replied:
^ This is causing the analyzer check failure that we're seeing. I still haven't tried ScaffoldMessenger yet myself, so it'll be interesting to see the new code :)

shihaohong replied:
Also, the docs are at master-api.flutter.dev: https://master-api.flutter.dev/flutter/material/ScaffoldMessenger-class.html

Since ScaffoldMessenger hasn't yet made it into `stable` yet.

Thesmader replied:
I changed the code to use ScaffoldMessenger to show the SnackBar. Please review.

Thesmader committed:
Refactor StatelessWidget content into build method

Removes custom stateless widgets and moves their content into the build
method. Adds duration property in the custom SnackBar example.

shihaohong commented on packages/flutter/lib/src/material/snack_bar.dart:
```diff
@@ -158,6 +158,69 @@ class _SnackBarActionState extends State<SnackBarAction> {
+///             label: "Action",
+///             onPressed: () {
+///               // Code to execute.
+///             }
```
nit: trailing comma
```suggestion
///             },
```

shihaohong commented on packages/flutter/lib/src/material/snack_bar.dart:
```diff
@@ -158,6 +158,69 @@ class _SnackBarActionState extends State<SnackBarAction> {
+///           ),
+///           content: Text("Awesome SnackBar!"),
+///           duration: Duration(milliseconds: 1500),
+///           width: 280.0, //Width of the SnackBar.
```
```suggestion
///           width: 280.0, // Width of the SnackBar.
```

shihaohong reviewed (approved):
LGTM, just some nitpicks

Thesmader committed:
Update packages/flutter/lib/src/material/snack_bar.dart

Co-authored-by: Shi-Hao Hong <shihaohong@google.com>

Thesmader committed:
Update packages/flutter/lib/src/material/snack_bar.dart

Co-authored-by: Shi-Hao Hong <shihaohong@google.com>

Thesmader commented:
@shihaohong The ```Windows tool_tests``` check failed for the second time. Is there anything I should be doing on my side? :sweat_smile:

shihaohong commented:
It says it is still running from my end. Lets give it a little more time

johnpryan commented:
@Thesmader Thanks for participating in AdoptAWidget. We would like to send you a special thanks. If you are interested, please send an email to nyener at google dot com and provide a link to this issue.
